include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -527,6 +527,29 @@ typedef struct OrtApiBase OrtApiBase;
 */
 ORT_EXPORT const OrtApiBase* ORT_API_CALL OrtGetApiBase(void) NO_EXCEPTION;
 
+/** \brief Thread work loop function
+*
+* Onnxruntime will provide the working loop on custom thread creation
+* Argument is an onnxruntime built-in type which will be provided when thread pool calls OrtCustomCreateThreadFn
+*/
+typedef void (*OrtThreadWorkerFn)(void* ort_worker_fn_param);
+
+typedef const struct OrtCustomHandleType{ char __place_holder; }* OrtCustomThreadHandle;
+
+/** \brief Ort custom thread creation function
+*
+* The function should return a thread handle to be used in onnxruntime thread pools
+* Onnxruntime will throw exception on return value of nullptr or 0, indicating that the function failed to create a thread
+*/
+typedef OrtCustomThreadHandle (*OrtCustomCreateThreadFn)(void* ort_custom_thread_creation_options, OrtThreadWorkerFn ort_thread_worker_fn, void* ort_worker_fn_param);
```

Comment:
Already completed, but we should spell out readable names in all public API's as either abbreviations or full words, not vowelless contractions :/ (`Function`, not `Fn`, unless we also want to call the other method <code>SetGlobalCustomThreadCreation<i>Optns</i></code>). Additionally these new functions are inconsistent with existing precedent in the header file which fully spells out the word <code>OrtLogging<i>Function</i></code>.